### PR TITLE
Add test 'Activate merchandise returns'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.7.x
| Description?      | Add test 'Activate merchandise returns'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no
| Possible impacts? | TEST_PATH="functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js" URL_FO=yourShopURL  npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22523)

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        page,
+        dashboardPage.ordersParentLink,
+        dashboardPage.ordersLink,
+      );
+
+      const pageTitle = await ordersPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(ordersPage.pageTitle);
+    });
+
+    it('should filter the Orders table by the default customer \'J. DOE\' and check the result', async function () {
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Should not be hardcoded, should be extracted from the customer data fixture.

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', `filterOrder${index}`, baseContext);
+
+      await ordersPage.filterOrders(page, 'input', 'customer', 'J. DOE');
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Same here

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+      const textColumn = await ordersPage.getTextColumn(page, 'customer', 1);
+      await expect(textColumn).to.contains('J. DOE');
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Same here

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+      const pageTitle = await viewOrderPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(viewOrderPage.pageTitle);
+    });
+
+    it('should change the order status to \'Shipped\' and check it', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', `updateOrderStatus${index}`, baseContext);
+
+      const result = await viewOrderPage.modifyOrderStatus(page, 'Shipped');
+      await expect(result).to.equal('Shipped');
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Shouldn't that be handled at the page class level ?

File: tests/UI/pages/BO/customerService/merchandiseReturns/index.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  constructor() {
+    super();
+
+    this.pageTitle = 'Merchandise Returns •';
+    this.successfulUpdateMessage = 'The settings have been successfully updated.';
+
+    // Selectors
+    // Options
+    this.generalForm = '#order_return_fieldset_general';
+    this.enableOrderReturnLabel = toggle => `${this.generalForm} label[for='PS_ORDER_RETURN_${toggle}']`;
```
Comment (SimonGrn) [inline tests/UI/pages/BO/customerService/merchandiseReturns/index.js:14]: This is an ugly selector.

File: tests/UI/pages/BO/orders/view.js [hunk 6]
```diff
@@ -262,6 +262,15 @@ class Order extends BOBasePage {
...
     /* eslint-enable no-return-assign, no-param-reassign */
   }
+
+  /**
+   * Is return products button visible
+   * @param page
+   * @returns {Promise<boolean>}
+   */
+  isReturnProductsButtonVisible(page) {
+    return this.elementVisible(page, '#order-view-page button.btn.btn-action.return-product-display');
```
Comment (SimonGrn) [inline tests/UI/pages/BO/orders/view.js]: Shouldn't that be a selector ?

File: tests/UI/pages/BO/customerService/merchandiseReturns/index.js (hunk 5 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/customerService/merchandiseReturns/index.js:14]: Why ? all our switch buttons selectors are like this one

File: tests/UI/pages/BO/orders/view.js (hunk 6 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/orders/view.js]: yes! I will fix it, thanks :+1:

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js (hunk 4 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Done thx :+1:

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Done thx :+1:

File: tests/UI/pages/BO/customerService/merchandiseReturns/index.js (hunk 5 above)
Comment (SimonGrn) [reply tests/UI/pages/BO/customerService/merchandiseReturns/index.js:14]: That's not really a problem and I guess Boubker will refactor it at some point in the future. Don't worry.

**Review by boubkerbribri — DISMISSED** [review]

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+      // Click on view my shop
+      page = await viewOrderPage.viewMyShop(page);
+
+      // Change FO language
+      await homePage.changeLanguage(page, 'en');
+
+      const isHomePage = await homePage.isHomePage(page);
+      await expect(isHomePage, 'Home page is not displayed').to.be.true;
+
+      if (index === 0) {
```
Comment (PierreRambaud) [inline tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js:140]:
```suggestion
      if (test.args.enable) {
```
It's more obvious than using an index you don't understand :sweat_smile:

File: tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+        const pageTitle = await myAccountPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(myAccountPage.pageTitle);
+      } else {
+        await homePage.goToYourAccountPage(page);
+
+        const pageTitle = await myAccountPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(myAccountPage.pageTitle);
+      }
+    });
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]:
```suggestion
    it('should go to FO', async function () {
    await testContext.addContextItem(this, 'testIdentifier', `goToFO${index}`, baseContext);

    // Click on view my shop
    page = await viewOrderPage.viewMyShop(page);

    // Change FO language
    await homePage.changeLanguage(page, 'en');

    const isHomePage = await homePage.isHomePage(page);
    await expect(isHomePage, 'Home page is not displayed').to.be.true;
  });

  // Go to My account page by login the first time and click on account link the second time
  if(index === 0) {
    it('should login and go to account page', async function () {
      await testContext.addContextItem(this, 'testIdentifier', `goToAccountPage${index}`, baseContext);

      await homePage.goToLoginPage(page);
      await foLoginPage.customerLogin(page, DefaultAccount);

      const isCustomerConnected = await foLoginPage.isCustomerConnected(page);
      await expect(isCustomerConnected).to.be.true;

      const pageTitle = await myAccountPage.getPageTitle(page);
      await expect(pageTitle).to.contains(myAccountPage.pageTitle);
    });
  } else {
    it('should go to account page', async function () {
      await testContext.addContextItem(this, 'testIdentifier', `goToAccountPage${index}`, baseContext);

      await homePage.goToYourAccountPage(page);

      const pageTitle = await myAccountPage.getPageTitle(page);
      await expect(pageTitle).to.contains(myAccountPage.pageTitle);
    });
  }
```

Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/05_customerService/03_merchandiseReturns/01_activateMerchandiseReturn.js]: Done thx :+1:

**Review by boubkerbribri — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh & @boubkerbribri

**Pull request merged**